### PR TITLE
DDF-4709 fixed session memory cleanup

### PR DESCRIPTION
**Pull request by ahoffer**
#### What does this PR do?
Fixes a memory leak. This PR is a cherry pick of a commit that will also be merged to the master branch. See https://github.com/codice/ddf/pull/4740
#### Who is reviewing it? 
@brjeter @stustison @clockard @troymohl

### CODE REVIEW

File: platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/security/session/AttributeSharingHashSessionIdManager.java [hunk 1]
```diff
@@ -82,10 +81,15 @@ public void invalidateSession(
 
       final Optional<String> sessionIdOptional =
           idManager
-              .latestSessionAttributes
-              .entrySet()
+              .dataStores
```
Comment (paouelle) [inline platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/security/session/AttributeSharingHashSessionIdManager.java]: ❗️ the `dataStores` list is being accessed from multiple threads either by iteration (as here) or to add or delete entries. However, I do not see any concurrency considerations which would require the collection to be synchronized on whenever accessed, modified or while being iterated.

**Comment by brjeter** [comment]
✅ Hero successful.
- I installed through the UI
- I logged in as multiple users across different contexts and tested different permutations of logging out of one context and accessing/logging into another
- I tested the `security:listsessions` and `security:endsession <user>` commands with various scenarios including multiple users.
- I tested with multiple restarts after logging in/out.
- I also tested the above with the hostname changed to non-localhost.

File: platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/security/session/AttributeSharingHashSessionIdManager.java [hunk 2]
```diff
@@ -183,25 +188,21 @@ protected void provideNewSessionAttributes(
       String id,
       Map<String, Object> sessionAttributes) {
     // Make sure these attributes are different than the latest.
-    if (sessionAttributes != null && !sessionAttributes.equals(getLatestSessionAttributes(id))) {
+    if (sessionAttributes != null) {
       LOGGER.trace("Pushing new session attributes to all web contexts for session {}", id);
-      latestSessionAttributes.put(id, sessionAttributes);
       dataStores
           .stream()
           .filter(ds -> ds != callingDataStore)
```
Comment (paouelle) [inline platform/platform-paxweb-jettyconfig/src/main/java/org/codice/ddf/security/session/AttributeSharingHashSessionIdManager.java]: ❓ Was the intent to exclude the calling data sotre by checking its identity and not equality?

**Comment by brjeter** [comment]
✅ Hero'd again after addressing @paouelle's comment.

**Pull request merged**